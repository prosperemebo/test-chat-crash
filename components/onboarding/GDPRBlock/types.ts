export interface GDPRBlockProps {
    isAgreedToTerms: boolean
    onAgreedToTerms: () => void
    isLoading: boolean
}