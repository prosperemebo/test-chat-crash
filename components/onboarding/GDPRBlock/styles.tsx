import styled from 'styled-components/native'
import { Image as ExpoImage } from 'expo-image'
import { Dimensions } from 'react-native'

const { width, height } = Dimensions.get('window')

export const Wrapper = styled.View`
    flex: 1;
    align-items: center;
    paddingHorizontal: 21px;
`

export const ImageWrapper = styled.View`
    height: ${height * 0.35}px;
    justify-content: center;
    align-items: center;
    margin-top: 30px;
`

export const TextWrapper = styled.View``

export const Image = styled(ExpoImage)`
    height: ${height * 0.3}px;
    width: ${height * 0.3}px;
    resize-mode: contain;
`

export const Header = styled.Text`
    font-size: 20px;
    font-family: 'Poppins-Bold';
    color: ${props => props.theme.onboarding.headerTextColor};
    margin-vertical: 10px;
    text-align: center;
`

export const Description = styled.Text`
    color: ${props => props.theme.onboarding.headerTextColor};
    font-size: 13px;
    margin-top: 10px;
    margin-bottom: 20px;
    text-align: center;
    font-family: 'Poppins-Regular';
    width: ${width * 0.85}px;
`

export const SwitchWrapper = styled.View`
    background-color: ${props => props.theme.backgroundColor2};
    border-color: ${props => props.theme.inActiveBorderColor2};
    border-width: 1.5px;
    border-radius: 10px;
    flex-direction: row;
    justify-content: space-between;
    padding: 14px 16px;
`

export const CheckBoxWrapper = styled.View`
    margin-top: 20px;
    margin-bottom: 80px;
    flex-direction: row;
`