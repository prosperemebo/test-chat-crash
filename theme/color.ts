export const WHITE = '#FFFFFF'
export const t10WHITE = 'rgba(255,255,255,0.1)'
export const t76WHITE = 'rgba(255,255,255,0.76)'

export const RED = '#f60004'
export const RED1 = '#D33E41'


export const BLACK = '#000000'
export const BLACK2 = '#050608'
export const nBLACK = '#15161A'
export const nBLACK10 = '#1F2025'
export const nBLACK30 = '#28292E'
export const nBLACK40 = '#050608'

export const nGRAY10 = '#6B6E79'
export const nGRAY20 = '#858995'
export const nGRAY25 = '#54575e'
export const nGRAY30 = '#2f3137'
export const nGRAY35 = '#52555D'
export const nGRAY40 = '#35363B'


export const tGRAY20 = '#85899599'
export const tGRAY30 = '#545458A6'
export const tGRAY40 = '#45464C'
export const GRAY40 = '#cfcfd0'
export const GRAY45 = '#c9c9cb'
export const GRAY50 = '#aaaaaa'
export const GRAY60 = '#52545D'
export const t40GRAY60 = '#52545D66'
export const GRAY70 = '#383B40'
export const GRAY90 = '#27292d'
export const GRAY100 = '#1f2023'
export const GRAY110 = '#151619'

export const GRAY_ONBOARD = '#8791a5'

export const BLUE = '#50b5e8'
export const BLUE2 = '#0A84FF'

export const PURPLE = '#9A1F99'

export const nBLUE = '#8D88A8'
export const nPURPLE = '#C47AF5'
export const nPURPLE10 = '#873FBB'
export const nPURPLE20 = '#873FBB'
export const nPURPLE40 = '#C882F8'

export const nORANGE10 = '#FF6705'
export const nORANGE20 = '#FF7905'
export const nORANGE40 = '#FFB700'

export const YELLOW2 = '#EBC789'
export const nYELLOW = '#F5D193'

export const nGRAYBACKGROUND2 = '#292A30'
export const nGRAYBACKGROUND3 = '#383940'
export const nGRAYBACKGROUND4 = '#989898'

export const gradient = {
  COLORS: [BLUE, PURPLE],
  START: { x: 0, y: 1 },
  END: { x: 1, y: 0 },
  LABEL_COLOR: WHITE,
}

export const orangeGradient = {
  COLORS: [nORANGE10, nORANGE40],
  START: { x: 0, y: 1 },
  END: { x: 1, y: 0 },
  LABEL_COLOR: WHITE,
}