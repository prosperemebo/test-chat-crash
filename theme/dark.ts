import {
  WHITE,
  BLACK,
  nGRAY10,
  GRAY_ONBOARD,
  nPURPLE,
  nBLUE,
  GRAY40,
  nGRAY20,
  RED1,
  tGRAY20,
  RED,
  nBLACK,
  nORANGE10,
  nORANGE40,
  nPURPLE10,
  GRAY70,
  tGRAY30,
  BLUE2,
  BLACK2,
  nGRAY25,
  nBLACK30,
  GRAY50,
  nBLACK10,
  t10WHITE,
  nGRAYBACKGROUND2,
  GRAY100,
  GRAY60,
  nGRAY40,
  nGRAY30,
  YELLOW2,
  nYELLOW,
  nGRAYBACKGROUND3,
  tGRAY40,
  nGRAY35,
  nBLACK40,
  t40GRAY60,
  nPURPLE20,
  nPURPLE40,
} from './color'

export const dark = {
  tint: nPURPLE,
  primaryTextColor: WHITE,
  primaryBg: nBLACK10,
  headerBg: GRAY100,
  headerBorderColor: GRAY60,
  secondaryTextColor: GRAY60,
  secondaryBgLight: nBLACK30,
  secondaryBg: nBLACK,
  serviceBg: GRAY100,
  inActiveBorderColor2: t10WHITE,
  backgroundColor2: nGRAYBACKGROUND2,
  backgroundColor3: nGRAYBACKGROUND3,
  backgroundColor4: nBLACK10,

  auth: {
    socialIconColor: WHITE,
    socialLoginButton: BLACK,
    dividerColor: nGRAY10,
    noticeText: WHITE,
    forgotPasswordLabel: WHITE,
  },

  onboarding: {
    headerTextColor: WHITE,
    headerStepsTextColor: GRAY_ONBOARD,
    progressBarColor: nPURPLE,
    progressBarBackgroundColor: nBLUE,
    selectedItemTextColor: WHITE,
    disabledItemTextColor: GRAY40,
    gameSectionOverlayColor: nPURPLE,
    gameEmptySectionColor: BLACK,
    gameEmptyIconColor: BLACK2,
  },

  common: {
    labelColor: WHITE,
    infoTextColor: nGRAY20,
    inputIconColor: nGRAY20,
    inputPlaceholderColor: nGRAY20,
    serviceTextColor: WHITE,
    inputTextColor: WHITE,
    inputTextError: RED1,
    inputBorderColor: tGRAY20,
    inputErrorBorderColor: RED,
    inputBackgroundColor: nBLACK,
    toastBackground: nBLACK,
    toastBorderColor: nPURPLE,
    toastTextColor: WHITE,
    buttonGradientOrangeStart: nORANGE10,
    buttonGradientOrangeEnd: nORANGE40,
    buttonGradientPurpleStart: nPURPLE40,
    buttonGradientPurpleEnd: nPURPLE20,
    buttonGradientGrayStart: nGRAY10,
    buttonGradientGrayEnd: nGRAY20,
    buttonDartBackgroundColor: BLACK,
    backButtonColor: nGRAY20,
    switchTextColor: WHITE,
    checkboxIconColor: nPURPLE,
    gradientButtonColorStart: nPURPLE10,
    gradientButtonColorEnd: nPURPLE,
    borderColor: GRAY70,
    genderPickerBorderColor: nGRAY25,
    genderPickerActiveTextColor: WHITE,
    genderPickerTextColor: GRAY50,
    checkboxTextColor: WHITE,
    iconButtonColor: WHITE,
    backgroundColor: nBLACK10,
  },

  settings: {
    socialIconBackgroundColor: nGRAY40,
    premiumFeatureTextColor: nPURPLE,
    buttonIconColor: nGRAY20,
    serviceItemBgColor: nGRAY30,
  },

  postOnboarding: {
    backgroundColor: nBLACK10,
    inputIconColor: nGRAY20,
    inputIconErrorColor: RED1,
    disclosureTextColor: nGRAY20,
    loadingIndicatorColor: BLACK,
  },

  profile: {
    backgroundColor: nBLACK10,
    improveBackgroundColor: YELLOW2,
    improveBannerBackgroundColor: nYELLOW,
    improveBannerTitleColor: nBLACK30,
    coverPictureCardRemoveButtonBg: tGRAY30,
    profilePicturesCountColor: tGRAY40,
  },

  chats: {
    iconColor: nGRAY20,
    inputColor: nBLACK40,
    placeholderColor: nGRAY35,
    appbarBorderColor: t40GRAY60,
    backgroundColor: nBLACK10,
  },

  nav: {
    bottomTabLabelBackgroundColor: nBLACK,
    bottomTabBackgroundColor: nBLACK,
  },

  other: {
    updateApp: {
      textBorderColor: tGRAY30,
      buttonTextColor: BLUE2,
      textColor: WHITE,
    },
  },
}
