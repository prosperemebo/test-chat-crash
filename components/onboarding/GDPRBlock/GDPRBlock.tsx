import { ImageWrapper, Wrapper, Image, TextWrapper, Header, Description, SwitchWrapper } from './styles'
import { SwitchInput } from '@/components/common'
import i18n from '@/localization/i18n'
import { GDPRBlockProps } from './types'

export const GDPRImage = require('@/assets/images/onboarding/gdpr.png')

const GDPRBlock = (props: GDPRBlockProps) => {
  const { isAgreedToTerms, onAgreedToTerms, isLoading } = props

  return <Wrapper>
    <ImageWrapper>
      <Image source={GDPRImage} />
    </ImageWrapper>
    <TextWrapper>
      <Header>{i18n.t('onb:gdprHeader')}</Header>
      <Description>{i18n.t('onb:gdprExplanation')}</Description>
    </TextWrapper>
    <SwitchWrapper>
      <SwitchInput
        label={i18n.t('onb:iAgreeTermsAndConditions')}
        value={isAgreedToTerms}
        onValueChange={onAgreedToTerms}
        type="gdpr"
        disabled={isLoading}
        isLoading={isLoading}
      />
    </SwitchWrapper>
  </Wrapper>
}

export default GDPRBlock