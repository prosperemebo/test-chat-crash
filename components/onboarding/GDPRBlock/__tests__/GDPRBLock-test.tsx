import { render } from '@testing-library/react-native'
import { ThemeProvider } from 'styled-components/native'
import { describe, expect, test } from '@jest/globals'
import GDPRBlock from '../'
import { dark } from '@/theme'

describe('<GDPRBlock />', () => {

  test('GDPRBlock renders correctly', () => {
    const tree = render(
      <ThemeProvider theme={dark}>
        <GDPRBlock
          isLoading={false}
          isAgreedToTerms={false}
          onAgreedToTerms={() => {}}
        />
      </ThemeProvider>,
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})