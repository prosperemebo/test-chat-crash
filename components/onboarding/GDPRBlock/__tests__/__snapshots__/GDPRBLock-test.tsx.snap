// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GDPRBlock /> GDPRBlock renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "paddingHorizontal": 21,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "height": 466.9,
        "justifyContent": "center",
        "marginTop": 30,
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "height": 400.2,
          "resizeMode": "contain",
          "width": 400.2,
        }
      }
    />
  </View>
  <View
    style={{}}
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontFamily": "Poppins-Bold",
          "fontSize": 20,
          "marginVertical": 10,
          "textAlign": "center",
        }
      }
    >
      MOCKED_STRING:onb:gdprHeader
    </Text>
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontFamily": "Poppins-Regular",
          "fontSize": 13,
          "marginBottom": 20,
          "marginTop": 10,
          "textAlign": "center",
          "width": 637.5,
        }
      }
    >
      MOCKED_STRING:onb:gdprExplanation
    </Text>
  </View>
  <View
    style={
      {
        "backgroundColor": "#292A30",
        "borderBottomLeftRadius": 10,
        "borderBottomRightRadius": 10,
        "borderColor": "rgba(255,255,255,0.1)",
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "borderWidth": 1.5,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 14,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 14,
      }
    }
  >
    <View
      style={
        {
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontFamily": "Poppins-Regular",
              "fontSize": 12,
              "fontStyle": "normal",
              "fontWeight": "300",
              "lineHeight": 22,
              "width": 625,
            }
          }
          type="gdpr"
        >
          MOCKED_STRING:onb:iAgreeTermsAndConditions
        </Text>
        <View
          style={
            {
              "flexDirection": "row",
              "width": 100,
            }
          }
        >
          <RCTSwitch
            accessibilityRole="switch"
            disabled={false}
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "height": 31,
                "width": 51,
              }
            }
            value={false}
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
